pontoon/base/utils.py

```diff
@@ -398,17 +386,75 @@ def get_download_content(slug, code, part):
             vcs_translation.update_from_db(entity)
 
     resource_file.save(locale)
+    return locale_path, source_path
+
+
+def get_download_content(slug, code, part):
+    """
+    Get content of the file to be downloaded.
+
+    :param str slug: Project slug.
+    :param str code: Locale code.
+    :param str part: Resource path or Subpage name.
+    """
+    from pontoon.base.models import (
+        Locale,
+        Project,
+        Resource,
+    )
+
+    project = get_object_or_404(Project, slug=slug)
+    locale = get_object_or_404(Locale, code__iexact=code)
+
+    # Download a ZIP of all files if project has less than 10 resources
+    if 1 < project.resources.count() < 10:
+        files = {}
+
+        for resource in project.resources.all():
+            locale_path, source_path = _generate_file_for_download(resource, locale)
+
+            if not locale_path:
+                return None, None
+
+            files[resource.path] = (locale_path, source_path)
+
+        s = StringIO.StringIO()
+        zf = zipfile.ZipFile(s, "w")
+
+        for fpath, value in files.iteritems():
```

Comment:
(Ideas for future) What would you say to move this into celery task which would generate this file after changes in translations? Zipping few big files can take some time and it will still be more faster if we generate them once for some time than per every request. We could also use some fancy CDN like cloudflare which could be benefit for end-users.